server/api/ballot_manifest.py

```diff
@@ -115,6 +119,36 @@ def process():
 
     process_file(session, file, process)
 
+    # If CVR file already uploaded, try reprocessing it, since it depends on
+    # batch names from the manifest
+    if jurisdiction.cvr_file:
+        logger.info(
+            f"START_REPROCESSING_CVRS {dict(election_id=jurisdiction.election.id, jurisdiction_id=jurisdiction.id)}"
+        )
+        # First, clear out the previously processed data.
+        CvrBallot.query.filter(
+            CvrBallot.batch_id.in_(
+                Batch.query.filter_by(jurisdiction_id=jurisdiction.id)
+                .with_entities(Batch.id)
+                .subquery()
+            )
+        ).delete(synchronize_session=False)
+        jurisdiction.cvr_contests_metadata = None
+        jurisdiction.cvr_file.processing_started_at = None
+        jurisdiction.cvr_file.processing_completed_at = None
+        jurisdiction.cvr_file.processing_error = None
+        # Because process_cvr_file uses a COPY command outside of the session
```

Comment:
oh booooooo. Are we sure we can't get COPY to run in the current transaction? Is it a sqlalchemy limitation?

(I understand this isn't a new issue, so we don't have to fix it now, but still makes me cry.)

Reply:
I know it's sad, but I really do think we're in ok shape here. If the CVR reprocessing fails, the file will get picked up and tried again by bgcompute. In fact, we don't even _need_ to call process_cvr_file here explicitly, though I think it's clearer to do so.

I can look into trying to run COPY in the current transaction, though I couldn't get it to work the first time around.